Add Z-Wave overview and setup sections (#26862)
home-assistant/home-assistant.github.io: source/_integrations/zwave_js.markdown

ha_zeroconf: true
---

> The Z-Wave integration allows you to control a Z-Wave network via the [Node Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.

Reviewer: ```suggestion
The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.
```
There's no "Node" in the official name, just in the URL (because it is a Node.js project)

## Overview

> The Z-Wave integration in Home Assistant talks to one of two possible add-ons: the official Z-Wave JS add-on, or the community add-on Z-Wave JS UI. Both add-ons include the Node Z-Wave JS driver. This driver talks to a controller device (Z-Wave dongle or GPIO (general purpose input/output) board)

Reviewer 2: It reads a bit weird that we say here that the integration talks to one of two add-ons, and then further below under the advanced instructions we say that you can also do without any add-on.

I think we should not talk about add-ons at all in the overview. I think we should just list the three requirements, a controller, the Z-Wave JS server and the Z-Wave integration.

Then in the next section we explain that the easiest way to get started running the server is to use the official add-on. The other ways of running the server are explained in the advanced installation instructions.

Author: Thank you for the input, @MartinHjelmare! I felt the need to introduce some of the terminology around Z-Wave, as I was confused about it during the installation procedure. But I realize that it is a lot of implementation detail to start with. I removed the section.

via USB or TCP.

<figure>
  <img src="/images/integrations/z-wave/z-wave-terminology.png" alt="Z-Wave terminology overview" style="width:100%">
  <figcaption>Z-Wave terminology overview</figcaption>
</figure>

To run a Z-Wave network, you need the following elements:

* A [supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).
* A running Z-Wave JS server (using only **one** of the add-ons described here).
* An installed Z-Wave integration in Home Assistant.

## Setting up a Z-Wave JS server

The easiest way to get started is by using the built-in Z-Wave JS add-on in Home Assistant.

> If you run Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Z-Wave JS Server add-on, refer to the [advanced installation instructions](#advanced-installation-instructions).

Reviewer: I don't think the addon is called "Z-Wave JS Server add-on"

## Setting up a Z-Wave JS server

The easiest way to get started is by using the built-in Z-Wave JS add-on in Home Assistant.

For other ways to setup a Z-Wave server, refer to the [advanced installation instructions](#advanced-installation-instructions).

Follow these steps:

1. Open the Home Assistant user interface.
1. Plug the Z-Wave dongle into the device running Home Assistant.
   * Most likely, your dongle will be recognized automatically. On the user interface, you will be asked if you want to set up this device with the Z-Wave JS add-on. Select **Submit**.
   * If your dongle is not recognized, follow these steps:

{% details "Manual setup steps" %}
Use this My button:

{% my config_flow_start badge domain="zwave_js" %}, or follow these steps:

* Browse to your Home Assistant instance.
* In the sidebar click on _**{% my config icon %}**_.
* From the configuration menu select: _**{% my integrations %}**_.
* In the bottom right, click on the
  _**{% my config_flow_start icon domain="zwave_js" %}**_ button.
* From the list, search and select _**"Z-Wave"**_.
* Follow the instructions on screen to complete the set up.

{% enddetails %}

> 3. Wait for the installation to complete.

Reviewer 3: Something happend with the numbering here

Author: I was trying to use a workaround. If I put 1. here, the numbering starts over:

![image](https://user-images.githubusercontent.com/38767475/230018212-52648f53-8442-466a-a755-d4eac76e64ef.png)

it seems that %details% breaks up the list. I was not able to find a way around that.

1. You are prompted for network security keys.
   * If you are using Z-Wave for the first time, leave all the fields empty and select **Submit**. The system will generate network security keys for you.
   * If this Z-Wave dongle has already been paired with secure devices, you need to enter the previously used network key as the S0 network key. S2 security keys will be automatically generated for you.
   * Make sure that you keep a backup of these keys in a safe place in case you need to move your Z-Wave dongle to another device. Copy and paste them somewhere safe.
1. Wait for the Z-Wave JS add-on to start up.
1. Once the installation is complete, the **Device info** of the Z-Wave controller is shown.
   * You successfully installed the Z-Wave integration and the Z-Wave JS add-on.
   * You can now add devices to the Z-Wave network.

<p class='note'>

> While your Z-Wave mesh is permanently stored on your dongle, the additional metadata is not. When the Z-Wave integration starts up the first time, it will interview your entire Z-Wave network. Depending on the number of devices paired with the Z-Wave dongle, this can take a while. You can speed up this process by manually waking up your battery-powered devices. Most of the time, this is a press on the button on those devices (see their manual). It is not necessary to exclude and re-include devices from the mesh.

Reviewer 3: ```suggestion
While your Z-Wave mesh is permanently stored on your dongle, the additional metadata is not. When the Z-Wave integration starts up the first time, it will interview your entire Z-Wave network. Depending on the number of devices paired with the Z-Wave dongle, this can take a while. You can speed up this process by manually waking up your battery-powered devices. Most of the time, this is a button press on those devices (see their manual). It is not necessary to exclude and re-include devices from the mesh.
```

ha_zeroconf: true
---

The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.

## Overview

To run a Z-Wave network, you need the following elements:

> * A [supported Z-Wave controller](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).

Reviewer 3: ```suggestion
* A [supported Z-Wave controller](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First-time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).
```

## Overview

To run a Z-Wave network, you need the following elements:

* A [supported Z-Wave controller](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).
* A running Z-Wave JS server.
* An installed Z-Wave integration in Home Assistant.

## Setting up a Z-Wave JS server

The easiest way to get started is by using the built-in Z-Wave JS add-on in Home Assistant.

For other ways to setup a Z-Wave server, refer to the [advanced installation instructions](#advanced-installation-instructions).

Everyone else, follow these steps:

1. Open the Home Assistant user interface.
1. Plug the Z-Wave dongle into the device running Home Assistant.
   * Most likely, your dongle will be recognized automatically. On the user interface, you will be asked if you want to set up this device with the Z-Wave JS add-on. Select **Submit**.
   * If your dongle is not recognized, follow these steps:

> {% capture name %}{{ include.name | default: page.title }}{% endcapture %}
> {% capture domain %}{{ include.domain | default: page.ha_domain }}{% endcapture %}
> 

Reviewer 3: ```suggestion
```

{% details "Manual setup steps" %}
Use this My button:

> {% my config_flow_start badge domain=domain %}, or follow these steps:

Reviewer 3: ```suggestion
{% my config_flow_start badge domain="zwave_js" %}, or follow these steps:
```

* Browse to your Home Assistant instance.
* In the sidebar click on _**{% my config icon %}**_.
* From the configuration menu select: _**{% my integrations %}**_.

> {% if page.ha_integration_type == 'helper' %}
> * In top of the screen click the tab: _**{% my helpers %}**_.
> * In the bottom right, click on the
>   _**{% my config_flow_start icon domain=domain title="Create helper" %}**_ button.
> {% else %}

Reviewer 3: ```suggestion
```

* In the bottom right, click on the

>   _**{% my config_flow_start icon domain=domain %}**_ button.

Reviewer 3: ```suggestion
  _**{% my config_flow_start icon domain="zwave_js" %}**_ button.
```

> {% endif %}

Reviewer 3: ```suggestion
```

> * From the list, search and select _**"{{ name }}"**_.

Reviewer 3: ```suggestion
* From the list, search and select _**"Z-Wave"**_.
```

The following features can be accessed from the integration configuration panel:

* **Add device:** Allows you to pre-provision a SmartStart device or start the inclusion process for adding a new device to your network.
* **Remove device:** Starts the exclusion process for removing a device from your network.
* **Heal network:** Forces your network to rediscover routes to the controller from each device. This is useful when devices or the controller have moved to a new location, or if you are having significant problems with your network, but it also generates a lot of network traffic and should be used sparingly.

> * **Third party data opt-in/out:** Allows you to opt-in or out of telemetry that the zwave-js project collects to help inform development decisions, influence manufacturers, etc. This telemetry is disabled by default and has to be opted in to be activated.

Reviewer 3: ```suggestion
* **Third-party data opt-in/out:** Allows you to opt-in or out of telemetry that the Z-Wave JS project collects to help inform development decisions, influence manufacturers, etc. This telemetry is disabled by default and has to be opted in to be activated.
```
